testing/cffi0/test_verify.py

```diff
@@ -2586,3 +2588,21 @@ def test_passing_large_list():
     arg = list(range(20000000))
     lib.passing_large_list(arg)
     # assert did not segfault
+
+def test_no_regen():
+    from cffi.verifier import Verifier, _caller_dir_pycache, _FORCE_GENERIC_ENGINE
+    import os
+    ffi = FFI()
+    # make sure the module name is unique
+    modulename = "_cffi_test_no_regen" + str(_FORCE_GENERIC_ENGINE)
+    ffi.cdef("double cos(double x);")
+    lib = ffi.verify('#include <math.h>', libraries=lib_m, modulename=modulename)
+    assert lib.cos(1.23) == math.cos(1.23)
+    # Make sure that recompiling the same code does not rebuild the C file
+    cfile = os.path.join(ffi.verifier.tmpdir, f"{modulename}.c")
+    assert os.path.exists(cfile)
+    os.unlink(cfile)
+    assert not os.path.exists(cfile)
+    lib = ffi.verify('#include <math.h>', libraries=lib_m, modulename=modulename)
+    assert lib.cos(1.23) == math.cos(1.23)
+    assert not os.path.exists(cfile)
```

Comment:
This is the tricky part of this PR. As @arigo commented in #137 
> If you want to kill the test_v*2.py files and replace them with a single test that checks no C code is regenerated, then I'm fine with it.

so I did that.